Nemo Connect/Sources/NCNetworkSessionManagerDelegate.m

```diff
@@ -98,15 +98,23 @@ - (void)URLSession:(NSURLSession *)session downloadTask:(NSURLSessionDownloadTas
     {
         float progress = (double)totalBytesWritten / (double)totalBytesExpectedToWrite;
 
-        // The OS calls it with totalBytesExpectedToWrite = -1
-        if (progress < 0)
+        if (totalBytesExpectedToWrite < 0)
         {
-            progress = 0;
+            // If it is -1, return with totalBytes
```

Comment:
Because the the system can give -1 as totalBytesExpectedToWrite so this is a fallback mechanism to give a chance for the host app to calculate the progress by on its own (if it has the totalBytesExpectedToWrite property it can calculate the progress by its own )